data-generation/src/main/scala/utils/Benchmark.scala

```diff
@@ -0,0 +1,97 @@
+package utils
+
+import java.time.Instant
+import scala.collection.mutable
+import scala.concurrent.duration.NANOSECONDS
+
+/** Benchmark
+ * Can be used like this:
+ * {{{
+ *   val b = Benchmark(ConsoleWriter, textPrefix="doStuff")
+ *   b.writeHeader
+ *   b.benchmarkSingle({spark.doStuff})
+ *   b.textPrefix = "SomeOtherFunction"
+ *   b.benchmarkAvg({spark.someOtherFunction}, 10)
+ *   b.benchmarkAvg({spark.lastFunction},5, "lastFunction")
+ *   b.close() // should be called if its a FileWriter
+ * }}}
+ * textPrefix is a mutable variable that sets the name for the output format
+ *
+ * @param writer    IoC, write to file or to console
+ * @param inSeconds if seconds is the preferred output unit of time (Could actually just be removed)
+ */
+case class Benchmark(writer: Writer, inSeconds: Boolean = false, var textPrefix: String = "unset") {
```

Comment:
```suggestion
case class Benchmark(writer: Writer, inSeconds: Boolean = false, textPrefix: String) {
```
Er det noen tilfeller der vi _ikke_ ønsker å sette dette? Tenker det kan være nice å force oss med å være ryddige med navngivning fra start

Reply:
ja, var litt usikker på den. Men jeg tenkte egentlig at du instansierer 1 objekt for hele benchmarken. Men kan sikkert heller tenke at man instansierer 1 objekt pr benchmark. Er med på den